pkg/apis/core/v1/defaults.go

```diff
@@ -162,6 +162,9 @@ func SetDefaults_Pod(obj *v1.Pod) {
 		enableServiceLinks := v1.DefaultEnableServiceLinks
 		obj.Spec.EnableServiceLinks = &enableServiceLinks
 	}
+	if obj.Spec.PreemptionPolicy == "" {
```

Comment:
I'm not sure if you should be updating the same obj.Spec like this.

EG check the output of /pkg/registry/core/pod/storage:go_default_test here: https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/74614/pull-kubernetes-bazel-test/1132861361513041920